core/store/src/trie/outgoing_metadata.rs

```diff
@@ -0,0 +1,212 @@
+use std::collections::{BTreeMap, VecDeque};
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_primitives::errors::StorageError;
+use near_primitives::types::{ShardId, StateChangeCause};
+use near_primitives::version::ProtocolFeature;
+use near_schema_checker_lib::ProtocolSchema;
+use near_vm_runner::logic::ProtocolVersion;
+
+use crate::TrieUpdate;
+
+use super::TrieAccess;
+
+/// Keeps metadata about receipts stored in the outgoing buffers.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct OutgoingMetadatas {
+    /// Metadata for each shard.
+    pub metadatas: BTreeMap<ShardId, OutgoingBufferMetadata>,
+    /// Receipts are grouped into groups and metadatas keep group-level information.
+    /// A new group is started when the size of the last group exceeds this threshold.
+    pub group_size_threshold: u32,
```

Comment:
nit: Maybe wrap it in a config struct? It's kind of similar to the CongestionControl struct that holds both the info and the config, and actually contains the business logic.

Reply:
Agree